logger/ringpop.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logger
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/uber-common/bark"
+	"io/ioutil"
+)
+
+// Options is used to set the ringpop logger implementation and the minimum
+// output level for each named logger.
+type Options struct {
+	// The underlying logger implementation to use.
+	Logger bark.Logger
+
+	Ringpop    Level
+	Gossip     Level
+	Suspicion  Level
+	Ring       Level
+	Membership Level
+	Damping    Level
+}
+
+// RingpopLogger is a factory for ringpop named loggers. Each named logger can
```

Comment:
I like the clear separation here of the ringpop vs default factory implementation. Makes it very easy to understand what loggers we have in ringpop.